### PR TITLE
Pass along tissue type from Lims to Scout config file

### LINKED ISSUE #497: Tissue type should be passed on from lims (is it there?) to Scout.
So that is saved and displayed in the case page.
Maybe the option is already there, I don't know. To be checked and eventually implemented.

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR should add a field `tissue_type` for each individual when creating a scout config file.
fix #497 

**How to prepare for test**:
- [x] install on hasta stage `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh pass_tissue_type2`
- [x] activate stage: `us`

**How to test**:
- [x] `cg upload scout -p maindrum`

**Expected test outcome**:
- [x] The load command should create a yaml config file containing a tissue type set to 'blood' for the case samples.

**Review:**
- [x] code approved by @patrikgrenfeldt 
- [x] tests executed by @henrikstranneheim 
- [x] "Merge and deploy" approved by @patrikgrenfeldt 
Thanks for filling in who performed the code review and the test!

This is a minor version update because it adds a new functionality.

### CODE REVIEW

**Comment by northwestwitch** [comment]
Rewrote the PR. I'm having some new issues with the formatting. From the failure messages I'm not really understanding what's wrong with these files:

image

**Comment by northwestwitch** [comment]
It worked, thanks @patrikgrenfeldt !! :)

File: cg/meta/upload/scoutapi.py [hunk 1]
```diff
@@ -64,6 +73,7 @@ def build_samples(self, analysis_obj: models.Analysis, hk_version_id: int = None
                 "sex": link_obj.sample.sex,
                 "bam_path": bam_path,
                 "mt_bam": mt_bam_path,
+                "tissue_type": lims_sample.get("source") or "unknown",
```
Comment (patrikgrenfeldt) [inline cg/meta/upload/scoutapi.py]:
```suggestion
                "tissue_type": lims_sample.get("source", "unknown"),
```
Shorthand version, not sure it is more clear though

Comment (northwestwitch) [reply cg/meta/upload/scoutapi.py]: It is, thanks!

**Comment by henrikstranneheim** [comment]
The tissue type is `'tissue_type': 'unknown'` ✔️ 
Screenshot 2020-01-29 at 09 40 38

However, is also says this `Could not fetch sample ACC5610A5 from LIMS: 404: Sample not found: ACC5610A5`. I do not know if this is due to that the LIMS stage is not updated and if this has anything to do with the code changes

**Comment by northwestwitch** [comment]
> The tissue type is `'tissue_type': 'unknown'` ✔️
> Screenshot 2020-01-29 at 09 40 38
> 
> However, is also says this `Could not fetch sample ACC5610A5 from LIMS: 404: Sample not found: ACC5610A5`. I do not know if this is due to that the LIMS stage is not updated and if this has anything to do with the code changes

That's great that you are testing this! Let me check in LIMS and and stage database and see what samples we have, could be the code of the fact that stage doesn't have bosscod data, as you say

**Comment by northwestwitch** [comment]
Good news is I haven't found that sample in LIMS stage, so we can exclude one hypothesis I think. 
Bad news is I don't know which case/family would be a good candidate to test this PR. The samples I find in LIMS stage are stage samples and I don't know how to connect them to cases (like bosscod) because this info is not in lims.. And when I look at this interface: 
image
For possible candidates (families recently uploaded on scout stage, for testing other stuff) then I don't find the samples in LIMS stage.
That's why I didn't test the PR myself when I first wrote it.
 You could perhaps ask @patrikgrenfeldt or @barrystokman if they are aware of a suitable case?

**Comment by henrikstranneheim** [comment]
@mayabrandi Do you know of a good test case?

**Comment by barrystokman** [comment]
LIMS stage is very poorly populated with test data, and for that reason I use prod LIMS for testing. I can do that because I only read from LIMS and therefore don't risk ***king up prod data :)

We do really need a way to update stage LIMS so we can use that instead, but I'm not sure how to do that.

**Comment by northwestwitch** [comment]
Ok! I don't have much knowledge of LIMS at the moment, but perhaps it is possible to copy a case from lims prod to lims stage, I can look into it as soon as I have time (next week). It would be good to replicate a part of LIMS prod in LIMS stage, I agree, but to test this and possibly other stuff we need just one case. @henrikstranneheim thanks for trying out this PR anyway! I'll try to fix something in the next days.

**Comment by mayabrandi** [comment]
Hej!

Its been on the todo for a long time, to set up a jobb that continuous copies lims prod to lims stage. I did start looking at is, but for some reason we didn’t do it after all. I don’t remember why. Maybe Kenny remembers? If not, ill dig into it agan. Would be really good to have stage up to date with prod now and then.

Regarding copying a single sample, that is not possible unfortunately. Or im almost 100% its not possible. I can double check with Gurpreet.

**Comment by northwestwitch** [comment]
Ah ok, Thanks!

**Comment by northwestwitch** [comment]
Conflicts solved now @henrikstranneheim!

**Comment by henrikstranneheim** [comment]
Great! I will test in the afternoon

**Comment by northwestwitch** [comment]
> Great! I will test in the afternoon

Thanks, let me know how it goes and if something needs fixing!

**Comment by henrikstranneheim** [comment]
**Expected test outcome**:
1. `cg upload scout -p maindrum`
Screenshot 2020-03-20 at 13 26 47
✅

**Comment by henrikstranneheim** [comment]
@northwestwitch Looks good! Give it a quick read, but I think it lloks good.

**Comment by northwestwitch** [comment]
Let's merge immediately before someone changes his/her mind!!!

image

**Comment by northwestwitch** [comment]
> @northwestwitch Looks good! Give it a quick read, but I think it lloks good.

Thank you for testing! 🙏

**Comment by henrikstranneheim** [comment]
@patrikgrenfeldt This one is ready for merge and deploy, whenever you have time.

**Comment by patrikgrenfeldt** [comment]
finally deployed on clinical-db and hasta. Great work @northwestwitch and the rest!

**Pull request merged**